be/src/runtime/memory/thread_mem_tracker_mgr.cpp

```diff
@@ -49,15 +52,22 @@ void ThreadMemTrackerMgr::cancel_fragment() {
     _check_limit = false; // Make sure it will only be canceled once
 }
 
-void ThreadMemTrackerMgr::exceeded() {
+void ThreadMemTrackerMgr::exceeded(int64_t size) {
     if (_cb_func != nullptr) {
         _cb_func();
     }
     _limiter_tracker_raw->print_log_usage(_exceed_mem_limit_msg);
 
     if (is_attach_query()) {
-        // TODO wait gc
-        cancel_fragment();
+        if (_is_process_exceed) {
```

Comment:
could sleep 100ms and then check if sys mem exceed limit until thread_wait_gc_max_milliseconds. So that if there are memory available, we could use it as soon as possible.

Reply:
fix